torch/nn/modules/module.py

```diff
@@ -193,13 +193,38 @@ def _apply(self, fn):
         for module in self.children():
             module._apply(fn)
 
-        for param in self._parameters.values():
+        def compute_should_use_data(tensor, tensor_applied):
+            if torch._has_same_tensorimpl_type(tensor, tensor_applied):
+                # If the new tensor has the same TensorImpl type as the existing tensor,
```

Comment:
this comment just explains the code exactly, which is already simple and readable.  How about commenting about why we do it this way?

Reply:
Fixed.